### PR TITLE
v0.34.0

### PR DESCRIPTION
**Pull request by rix0rrr**
See CHANGELOG

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+* **event-targets:** `targets.EcsEc2Task` renamed to `targets.EcsTask`
+* * SNS - Subscription `endpoint` is now type `string` (previously `any`)
+* Step Functions - `result` in the Pass state is now type `map` (previously `any`)
+* the following modules are no longer released: `@aws-cdk/applet-js`, `@aws-cdk/aws-autoscaling-api`, `@aws-cdk/aws-codedeploy-api`
+* **cli:** applets are no longer supported as an app type, use "decdk" instead.
+* **core:** Properties passed to `addPropertyOverride` should match in capitalization to the CloudFormation schema (normally pascal case). For example, `addPropertyOverride('accessControl', 'xxx')` should now be `addPropertyOverride('AccessControl', 'xxx')`.
+
+
+
 # Change Log
```
Comment (RomainMuller) [inline CHANGELOG.md]: Isn't this meant to be at the very top of the file?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,67 @@
...
+* **s3:** default to KMS if encryptionKey is specified ([#2719](https://github.com/awslabs/aws-cdk/issues/2719)) ([ae4a04f](https://github.com/awslabs/aws-cdk/commit/ae4a04f)), closes [#2714](https://github.com/awslabs/aws-cdk/issues/2714)
+* **tokens:** enable type coercion ([#2680](https://github.com/awslabs/aws-cdk/issues/2680)) ([0f54698](https://github.com/awslabs/aws-cdk/commit/0f54698)), closes [#2679](https://github.com/awslabs/aws-cdk/issues/2679)
+
+
+### BREAKING CHANGES
+
+* **route53:** `recordValue: string` prop in `route53.TxtRecord` changed to `values: string[]`
+* `recordValue` prop in `route53.CnameRecord` renamed to `domainName`
+* `route53.AliasRecord` has been removed, use `route53.ARecord` or `route53.AaaaRecord` with the `target` prop.
+* **core:**
```
Comment (eladb) [inline CHANGELOG.md]: Remove this line

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+* **kms:** The `EncryptionKeyAlias` class was renamed to `Alias`.
+Associated types (such as `EncryptionKeyAliasProps`) were renamed in the
+same way.
+* **core:** * **cli:** This release requires CDK CLI >= 0.34.0
```
Comment (eladb) [inline CHANGELOG.md]: Clean up

**Comment by rix0rrr** [comment]
Build is broken atm, need a patch on jsii before we can merge this release.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+* **rds:** add engineVersion to DatabaseCluster ([#2698](https://github.com/awslabs/aws-cdk/issues/2698)) ([0593d51](https://github.com/awslabs/aws-cdk/commit/0593d51)), closes [#2212](https://github.com/awslabs/aws-cdk/issues/2212)
+* **rds:** add support for database instances ([#2187](https://github.com/awslabs/aws-cdk/issues/2187)) ([b864041](https://github.com/awslabs/aws-cdk/commit/b864041)), closes [#2075](https://github.com/awslabs/aws-cdk/issues/2075) [#1693](https://github.com/awslabs/aws-cdk/issues/1693)
+* **route53:** improve constructs for basic records ([#2741](https://github.com/awslabs/aws-cdk/issues/2741)) ([696f53f](https://github.com/awslabs/aws-cdk/commit/696f53f))
+* **s3:** default to KMS if encryptionKey is specified ([#2719](https://github.com/awslabs/aws-cdk/issues/2719)) ([ae4a04f](https://github.com/awslabs/aws-cdk/commit/ae4a04f)), closes [#2714](https://github.com/awslabs/aws-cdk/issues/2714)
+* **tokens:** enable type coercion ([#2680](https://github.com/awslabs/aws-cdk/issues/2680)) ([0f54698](https://github.com/awslabs/aws-cdk/commit/0f54698)), closes [#2679](https://github.com/awslabs/aws-cdk/issues/2679)
+
+
+### BREAKING CHANGES
```
Comment (jogold) [inline CHANGELOG.md]: The following breaking changes from #2187 are missing (were not included in the final PR description):
* `rds.RotationSingleUser` renamed to `rds.SecretRotation`
* `rds.ClusterParameterGroup` no longer have `setParameter` and `removeParameter` methods, use the `parameters` prop directly in the constructor instead.

**Pull request merged**